webauthn-authenticator-rs/src/MacAuthn/Sources/MacAuthn/MacAuthn.swift

```diff
@@ -0,0 +1,194 @@
+            let clientDataJSON = credential.rawClientDataJSON.toBase64Url()
+            let authenticatorData = credential.rawAuthenticatorData.toBase64Url()
+            let rawId = credential.credentialID.toBase64Url()
+            self.result = .ok([
+                "id": rawId,
+                "rawId": rawId,
+                "type": "public-key",
+                "response": [
+                    "clientDataJSON": clientDataJSON,
+                    "authenticatorData": authenticatorData,
+                    "signature": signature
+                ]
+            ])
+        } else if let credential = authorization.credential as? ASAuthorizationPlatformPublicKeyCredentialRegistration {
+            let rawId = credential.credentialID.toBase64Url()
+            let clientDataJSON = credential.rawClientDataJSON.toBase64Url()
+            let attestationObject = credential.rawAttestationObject!.toBase64Url()
+            self.result = .ok([
+                "id": rawId,
+                "rawId": rawId,
+                "type": "public-key",
+                "response": [
+                    "clientDataJSON": clientDataJSON,
+                    "attestationObject": attestationObject
+                ]
+            ])
+        } else if let credential = authorization.credential as? ASAuthorizationPlatformPublicKeyCredentialAssertion {
+            let signature = credential.signature.toBase64Url()
+            let clientDataJSON = credential.rawClientDataJSON.toBase64Url()
+            let authenticatorData = credential.rawAuthenticatorData.toBase64Url()
+            let rawId = credential.credentialID.toBase64Url()
+            self.result = .ok([
+                "id": rawId,
+                "rawId": rawId,
+                "type": "public-key",
+                "response": [
+                    "clientDataJSON": clientDataJSON,
+                    "authenticatorData": authenticatorData,
+                    "signature": signature
+                ]
+            ])
+        } else {
+            self.result = .error("unhandled credential")
+        }
+        NSApplication.shared.stop(0)
+    }
+    
+    func authorizationController(controller: ASAuthorizationController, didCompleteWithError error: Error) {
+        self.result = .error(error.localizedDescription)
+        NSApplication.shared.stop(0)
+    }
+}
+
+// ASAuthorizationController expects an ASPresentationAnchor (a NSWindow)
+// in order to know where to place itself. This function can create such
+// a window and run the NSRunLoop event loop to completion for it.
+func run(authController: ASAuthorizationController) -> String {
+    NSApplication.shared.setActivationPolicy(.regular)
+    let window = NSWindow(contentRect: NSMakeRect(0, 0, 1, 1), styleMask: .borderless, backing: .buffered, defer: false)
+    window.center()
+    window.makeKeyAndOrderFront(window)
+    
+    let applicationDelegate = ApplicationDelegate(window: window, authController: authController)
```

Comment:
Could this section cause conflicts if there's already a running `NSApplication` for the process?